lib/internal/checkUrl.js

```diff
@@ -66,6 +83,12 @@ function checkUrl(link, baseUrl, options, callback)
 		}
 		
 		linkObj.clean(link);
+		if (cache)
+		{
+			cache.store(link.url.resolved, function(cb){
```

Comment:
Since callback was defined at the top of the function, I thought that might cause confusion despite working. I'd like to avoid having the same variable redefined in nested functions unless you think it would be necessary. I've renamed it to `cacheCallback`. Let me know if you hate that one.